### PR TITLE
Merge dev into feat antiflood

### PR DESCRIPTION
**Pull request by bogdan-rosianu**

### CODE REVIEW

File: consensus/spos/worker_test.go [hunk 1]
```diff
@@ -818,82 +818,6 @@ func TestWorker_ProcessReceivedMessageShouldErrIfFloodIsDetectedOnTopic(t *testi
 	assert.Equal(t, expectedErr, err)
 }
 
-func TestWorker_ProcessReceivedMessageWrongHeaderShouldErr(t *testing.T) {
```
Comment (iulianpascalau) [inline consensus/spos/worker_test.go]: Why did you remove this test? It appears both in development and feat/antiflood?

Comment (bogdan-rosianu) [reply consensus/spos/worker_test.go]: it is already in the file. it just moved to the bottom of it

Comment (iulianpascalau) [reply consensus/spos/worker_test.go]: right
